### PR TITLE
Cert profile page dev merge

### PR DESCRIPTION
**Pull request by JackSMunn**
## Description

Refactored profile page, added edit profile, also changed around a ton of files, along with generalizing a bunch of files. Created error service.

## Changes

- fix(LogOutButton): logout button restyled
- fix(ProfileNameCircle): profileName has been deleted and replaced with ProfileNameCircle. It now only shows the initials of the user
- feat(ProfilePageNavigationButton): new component made for buttons on the profile page
- feat(UserInfo): new component made for the top part of new profile page (shows ProfileNameCircle, name, email, points)
- fix(Profile): profile page has been edited according to the figma
- new colors have been added to tailwind.config

Removed gradient from:
- EducadoModal
- Profile page
- Profile settings page
- Loading screen

Added general form components
- FormButton
- FormFieldAlert
- FormTextAlert
- PasswordField
- PasswordEye

- StorageService now provides set and get for both JWT and userInfo.
- Moved error messages into an error alert service.
- For now it's only used on the ChangePassword page, but it can be used anywhere to provide standard error messages.
- Some of these errors are a bit rough, and as such, caution should be taken in using them - just an easy way to provide feedback to the user.
- fix: updated snapshots
- fix(userApi.js):
- axios.patch changed to client.patch
- optional chaining used in error handling
- feat(FormButton): now also has a warning style (yellow button)
- feat(ChangePasswordModal): button changed to warning style
- feat(ProfileSettings):
- styling of the edit profile page
- checks whether any fields are changed, if not 'save' button disabled
- saveUserInfo changes user info to what is written in fields (rn only works if all fields are changed)
- all fields have to be changed in order for it to work (this probably has to do with the ensureNewValues function in the backend)
- information can only be changed once per login session?
- currently no validation on email field
- also no validation for zero character names
- 'salvar' button needs to be wider
- updated user information needs to be persisted across entire application, not just ProfileSettings (without having to log out and in)
- better error handling in general
- fix(ProfileSettings->EditProfile): ProfileSettings renamed to EditProfile
- feat(ProfileNameCircle): looksbetter
- feat(EditProfile):
- proper validation on all fields
- can change individual fields
- Fixed the back button issue on edit profile page. Also the change password page and edit profile page now have the same layout (minus the space between fields, those are still different).
- added test for edit profile navigation
- added comments
- feat(EditProfile.test): tests have been made
- changed all tests regarding hardcoded ports to not use the hardcoded ports
- added branch to actions

## Related Issues

None. 

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)
image
image
image
## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Stable branch: Dev
commit id:  5306d65

## Notes for Reviewers

We have outcommented some testing, that tested whether or not we are running from localhost, which is fairly redundant.

### CODE REVIEW

**Review by Donhusum — COMMENTED** [review]
I think it all looks pretty good

**Review by sofiiagran — CHANGES_REQUESTED** [review]
I get errors in the console when navigating to the profile page:  ERROR  Warning: Failed prop type: Invalid prop `style` of type `array` supplied to `FormButton`, expected `object`.
FormButton@http://192.168.0.224:19000/node_modules/expo/AppEntry.bundle:109020:90
    in NativeWind.FormButton

**Review by PrometheusBork — CHANGES_REQUESTED** [review]
Looks very nice! But currently the name and last name overflow doesn't work properly for me. It should say TobyOneTwoThreeFour Kenobi but "ur" is cut off, otherwise everything seems to work
image

**Review by PrometheusBork — APPROVED** [review]
Nice Work!

File: eml/components/profileSettings/ChangePasswordModal.jsx
Comment (amaliedilling) [inline eml/components/profileSettings/ChangePasswordModal.jsx:1]: remember to comment translations in English for all Portuguese text. not necessary for approval, but nice for better readability

File: eml/components/login/RegisterForm.js [hunk 1]
```diff
@@ -175,27 +179,6 @@ export default function RegisterForm() {
     }
   }
 
-  /**
-   * Stores the user info in async storage
-   * @param {*} userInfo: {id, firstName, lastName, email, completed courses}
-   */
-  async function saveUserInfoLocally(userInfo) {
```
Comment (amaliedilling) [inline eml/components/login/RegisterForm.js:182]: Is it safe to delete this?

File: eml/components/login/LoginForm.js [hunk 2]
```diff
@@ -33,28 +30,6 @@ export default function LoginForm() {
   // State variable to track password visibility
   const [showPassword, setShowPassword] = useState(false);
 
-  /**
-   * Stores the user info in async storage
-   * @param {*} userInfo: {id, firstName, lastName, email}
-   */
-  async function saveUserInfoLocally(userInfo) {
```
Comment (amaliedilling) [inline eml/components/login/LoginForm.js:40]: Is it safe to delete this? We still need to save some user info locally?

**Review by amaliedilling — COMMENTED** [review]
This screen isn't responsive to the emulator I test on (iOS tho so not necessary for approval)
image

File: eml/components/login/RegisterForm.js (hunk 1 above)
Comment (MaizeHH) [reply eml/components/login/RegisterForm.js:182]: the deleted function has the same functionality as setUserInfo (line 169), which has been imported from StorageServices

File: eml/components/login/LoginForm.js (hunk 2 above)
Comment (MaizeHH) [reply eml/components/login/LoginForm.js:40]: the deleted function has the same functionality as setUserInfo (line 56), which has been imported from StorageServices

**Review by ITPini — APPROVED** [review]

**Pull request merged**